x-pack/plugins/event_log/generated/mappings.json

```diff
@@ -292,6 +292,12 @@
                                                 "number_of_triggered_actions": {
                                                     "type": "long"
                                                 },
+                                                "total_alerts": {
+                                                  "type": "long"
+                                                },
+                                                "total_hits": {
```

Comment:
"hits" isn't a general alerting thing, so I think we'll need a comment somewhere describing what it is - perhaps in the `mappings.js` file?

Reply:
If "hits" isn't a general alerting thing, I don't think it should be part of the event-log. How do the "hits" differ from the "alerts"?

Reply:
Please see these two issues for details https://github.com/elastic/kibana/issues/120678 https://github.com/elastic/kibana/issues/120668, but gist here is `total_hits` is the total number of candidate alerts found during a specific execution, and `total_alerts` is the total number of those candidate alerts that were actually created during that execution. The totals between candidates and created can differ as a result of reaching the configured `max_signals` threshold, or if duplicate alerts (alerts previously detected) were identified, but never written (and I believe a case with exceptions as well, but would need to verify).

I totally agree about naming (perhaps `total_candidate_alerts` better fits?), but that aside, is there no representation of these concepts in general alerting? Is there no circuit breaker for the number alerts that can be written per execution, or management of duplicates where discerning between these two values is of use to the general alerting user (like for the security solution user outlined in the issues above)?

Reply:
Ah, interesting.  We'll have an issue open soon-ish (from some previous research) about having a maximum number of actions and alerts (separate values) per rule execution.  And we would want to track the total number as being different if we cap the number via the maximums.

So it isn't a thing now, but probably will be.  So ... maybe just some word-smithing on this.  Candidate doesn't seem quite right.  Something like "total" as the number the rule produced, and "triggered" instead of "candidate", so "totalAlerts" and "triggeredAlerts"?

Reply:
> If "hits" isn't a general alerting thing, I don't think it should be part of the event-log

We already have a few metrics stored in the event log which are not a general alerting thing (strictly speaking) but which make sense from the Security Solution and AAD perspective. These are `total_indexing_duration_ms`, `total_search_duration_ms` and `execution_gap_duration_s`. Imho the new metrics have the same relation to the overall logic of Detection Engine and AAD.

I agree that additional documentation and/or better naming would help though.

Reply:
Thanks for the explanation, @spong, that makes a lot of sense to me.

What about `total_alerts_created` and `total_alerts_detected`?

Reply:
> What about `total_alerts_created` and `total_alerts_detected`?

Those two sound good to me @kobelb! 🙂 I'll wait a moment to see if anyone else has additional input and will then make the change.

In addition to adding documentation within `event-log` as @pmuellr recommended, are there any other supporting docs we should be updating with these fields as well? I know there's the AAD schema spreadsheet, but I don't believe the `event-log` is schema represented in there, no?

Reply:
> In addition to adding documentation within event-log as @pmuellr recommended, are there any other supporting docs we should be updating with these fields as well? I know there's the AAD schema spreadsheet, but I don't believe the event-log is schema represented in there, no?

I'm not aware of any docs about the fields in the event-log or any docs that recommend that users search these indices. We've treated them as a traditional hidden index where users might find it, but they can't rely on the names of fields.

Reply:
Pulled this feature out to a dedicated PR (https://github.com/elastic/kibana/pull/126210), will ping reviewers when ready for review.